opencog/atoms/distvalue/CTHist.cc

```diff
@@ -0,0 +1,399 @@
+/*
+ * Copyright (C) 2019 SingularityNet
+ * All Rights Reserved
+ *
+ * Written by Roman Treutlein
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License v3 as
+ * published by the Free Software Foundation and including the exceptions
+ * at http://opencog.org/wiki/Licenses
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program; if not, write to:
+ * Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include <algorithm>
+#include <cmath>
+
+#include <opencog/util/exceptions.h>
+
+#include <opencog/atoms/distvalue/CTHist.h>
+
+using namespace opencog;
+
+template <typename val_t>
+void CTHist<val_t>::insert(const DVec & pos,const val_t & value)
+{
+	insert(CoverTreeNode<val_t>(pos,value));
+}
+
+template <typename val_t>
+void CTHist<val_t>::insert(const CoverTreeNode<val_t> & x)
```

Comment:
This and all of the other methods should be documented.  In addition, there should 
be a general description of what a cover-tree is, and why it is needed, and how it is used.
(I use histograms all the time in the language-learning code, and can't even begin to
imagine why a cover tree is needed)